Sources/SwiftInspectorVisitors/FunctionDeclarationVisitor.swift

```diff
@@ -3,16 +3,44 @@ import SwiftSyntax
 public final class FunctionDeclarationVisitor: SyntaxVisitor {
   public override func visit(_ node: FunctionDeclSyntax) -> SyntaxVisitorContinueKind {
     let name = node.identifier.withoutTrivia().description
+
     let functionSignatureVisitor = FunctionSignatureVisitor()
     functionSignatureVisitor.walk(node)
+
+    let modifiersVisitor = DeclarationModifierVisitor()
+    if let modifiers = node.modifiers {
+      modifiersVisitor.walk(modifiers)
+    }
+
     let info = FunctionDeclarationInfo(
+      modifiers: modifiersVisitor.modifiers,
       name: name,
       arguments: functionSignatureVisitor.arguments,
       returnType: functionSignatureVisitor.returnType)
     functionDeclarations.append(info)
     return .skipChildren
   }
 
+  public override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
```

Comment:
this change could be pulled out into a separate PR, but I realized our `FunctionDeclarationVisitor` was likely visiting functions from nested classes, and we really don't want it doing that (I can write a test for this before merging but wanted to get this thought down before I forgot 😅)

Reply:
might be good for me to pull this out into its own PR. I originally put this code here as a perf improvement and then realized that we likely have a bug here.

Reply:
gonna revert this here and instead fix this in https://github.com/fdiaz/SwiftInspector/pull/131